src/test/scala/chiselTests/QueueFlushSpec.scala

```diff
@@ -0,0 +1,181 @@
+package chiselTests
+
+import org.scalacheck._
+
+import chisel3._
+import chisel3.testers.BasicTester
+import chisel3.util._
+import chisel3.util.random.LFSR
+
+class ThingsPassThroughFlushQueueTester(elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+  
+  val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))  
+  q.io.flush.get := false.B
+  val elems = VecInit(elements.map(_.U))
+
+  val inCnt = Counter(elements.length + 1)
+  val outCnt = Counter(elements.length + 1)
+
+  q.io.enq.valid := (inCnt.value < elements.length.U)
+  q.io.deq.ready := LFSR(16)(tap)
+
+  q.io.enq.bits := elems(inCnt.value)
+  when(q.io.enq.fire()) {
+    inCnt.inc()
+  }
+  when(q.io.deq.fire()) {
+    //ensure that what comes out is what comes in
+    assert(elems(outCnt.value) === q.io.deq.bits)
+    outCnt.inc()
+  }
+  when(outCnt.value === elements.length.U) {
+    stop()
+  }
+}
+
+class QueueGetsFlushedTester (elements: Seq[Int], queueDepth: Int, bitWidth: Int, tap: Int, useSyncReadMem: Boolean) extends BasicTester {
+    val q = Module(new Queue(UInt(bitWidth.W), queueDepth, hasFlush = true))
+    val elems = VecInit(elements.map(_.U))
+
+  val inCnt = Counter(elements.length + 1)
+  val outCnt = Counter(elements.length + 1)
+  
+  val halfCnt = (queueDepth + 1)/2
+  val cycleCounter = Counter(elements.length + 1)
+  //testing a flush when 
+  val flush = LFSR(16)((tap + 3) % 16)
+  q.io.flush.get := flush
+  q.io.enq.valid := (inCnt.value < elements.length.U)
+  q.io.deq.ready := LFSR(16)(tap)
+  cycleCounter.inc() //counts every cycle
+
+  q.io.enq.bits := elems(inCnt.value)
+  when(q.io.enq.fire()) {
+    inCnt.inc()
+  }
+  when(q.io.deq.fire()) {
+    outCnt.inc()
+  }
+
+  when(flush) {
+    //check that queue gets flushed
+    assert((q.io.count === 0.U) || q.io.deq.valid, s"in count: ${q.io.count} and queue length: $queueDepth" ) 
+    assert(!q.io.enq.ready, s"in count: ${q.io.count} and queue length: $queueDepth") 
```

Comment:
can you also clean up these assert strings to be a bit more descriptive, or else remove them if they're just for debug? Might be better to say something like

```suggestion
    assert((q.io.count === 0.U) || q.io.deq.valid, p"Expected to be able to dequeue a flushed queue if it had elements prior to the flush, but got dequeue valid ${q.io.deq.valid}." ) 
    assert(!q.io.enq.ready, p"Expected enqueue to not be ready when flush is active, but got enqueue ${q.io.enq.ready}.") 
```